keps/sig-storage/20180515-svcacct-token-volumes.md

```diff
@@ -159,6 +162,34 @@ sources:
     audience: ca.istio.io
 ```
 
+### File Permission
+
+The service account secret was mounted as world readable(0644), so it is not an
+issue for non-root containers. However, it is no longer the case when switching
+to projected token volume, because it is mounted as root(uid=0) with `rw------`
+(0600). Containers running as non-root will not have access to the token volume.
+
+Setting the ownership of the volume properly can fix the issue above:
+1. If there is only one UserID (specified by `runAsUser`) shared by all the
+containers in a pod, set the owner of the projected service account volume to
+the UserID and set the mode to 0600.
+2. If `fsGroup` is specified, set the owner to be root and GroupID specified by
+`fsGroup`
+3. Otherwise, world readable 0644.
+
+#### Alternatives:
+1. create a volume for each UserID and set the owner to be that UserID with mode
+0600
+		+ if user doesn't specify `runAsUser`, fetching UserID in image requires a
+re-design of kubelet regarding volume mounts and image pulling.
+    + projected service account token is a pod level resource
+    + scalability issue. # of volumes = # of user IDs.
+2. set group ownership with mode 0640
+    + difficult to find a random unused GroupID
+    + not as fine-grained as UID.
+3. world read permission 0644 which is the same as the mode of secret
+    + security compromise
```

Comment:
Can you better define what you mean here? Because this case is what will end up being used if users don't specify `spec.securityContext.runAsUser` or `spec.containers[].securityContext.runAsUser` and also don't specify `spec.securityContext.fsGroup`, which I personally think will be more than 90% of pods.

Reply:
PTAL